src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -3734,6 +3810,124 @@ private AssociativeNode DFSEmitSplitAssign_AST(AssociativeNode node, ref List<As
             return node;
         }
 
+        /// <summary>
+        /// Converts lhs ident lists to a function call
+        /// a.x = 10 
+        ///     -> t = a.%set_x(10)
+        ///     
+        /// a.x.y = b + c 
+        ///     -> a.x.%set_y(b + c)
+        ///     
+        /// a.x[0] = 10
+        ///     ->  tval = a.%get_x()
+        ///         tval[0] = 10         
+        ///         tmp = a.%set_x(tval)
+        /// </summary>
+        /// <param name="astList"></param>
+        /// <returns></returns>
+        private List<AssociativeNode> TransformLHSIdentList(List<AssociativeNode> astList)
+        {
+            if (null != astList && astList.Count > 0)
+            {
+                List<AssociativeNode> newAstList = new List<AssociativeNode>();
+                foreach (AssociativeNode node in astList)
+                {
+                    BinaryExpressionNode bNode = node as BinaryExpressionNode;
+                    if (bNode == null)
+                    {
+                        newAstList.Add(node);
+                    }
+                    else
+                    {
+                        bool isLHSIdentList = bNode.LeftNode is IdentifierListNode;
+                        if (!isLHSIdentList)
+                        {
+                            newAstList.Add(node);
+                        }
+                        else
+                        {
+                            IdentifierNode lhsTemp = new IdentifierNode(Constants.kTempVar);
```

Comment:
Question: as all identifier list nodes will go through this transformation, and they share a same temporary variable, will this cause some update issue?